### DISCUSSION TITLE
Поддержка в играх

### DISCUSSION THREAD

**Discussion by bodun56** [vsulako/AFFBWheel#30] [General]
Предлагаю здесь делиться опытом, где и как, у кого какие грабли бывали или же есть.

У меня возник вопрос по поводу поведения в играх, таких как:
Project cars 1 и 2 части
NFS Shift 2
В данных играх, при подключенном контроллере, почему-то игра подзависает, хотя в настройках видит, определяет и назначает
Но при старте гонки, происходит зависание, до тех пор пока не отключить леонарду, затем отвисает и гонка стартует - но без меня

Forza Horizon 4 - здесь вообще игнорируется контроллер

В других же играх, всё отлично, всё работает и обратная связь в том числе
Вопрос в том, что только у меня ли так или может у кого-то ещё?

**Comment by vsulako** [comment]
Да, в некоторых играх имеется такое.
FH4, похоже, работает только с известными ей контроллерами по их протоколам.
Решение - эмулятор руля [GIMX](https://gimx.fr/).

**Reply by lazenyuk-dmitry** [reply]
Кстати, пока писал про прошивку под Logitech g25 кое что вспомнил. Я искал прошивки и в частности про эмуляцию g25 там сразу я так понял разработчик из России сделал какой то свой клон g25 прям платы печатал, а потом так же кто то из России сделал порт прошивки на Arduino Leonardo) а я в глубинах интернета где то находил комментарий на англоязычном форуме, типа можно ли сделать порт прошивки на Arduino и ему ответили что нельзя😄

**Comment by lazenyuk-dmitry** [comment]
Привет. Уже обсуждалось это. https://github.com/vsulako/AFFBWheel/issues/5#issuecomment-1112976693 вот здесь, с этого комментария, можешь почитать. Решили мы тогда что лучше Gimx использовать для поддержки в некоторых играх. Да и не дорого Gimx собрать. А добавление в этот проект поддержки или режима руля G27 может создать много проблем. Основная проблема что на Arduino Leonard мало память, тоесть сейчас проект уже занимает практически 100% памяти. А вообще есть проект прошивка для Arduino Leonard которая эмулирует Logitech g25 или g27. Есть группа в ВК там недавно это обсуждали, она ещё жива, могу дать ссылку на группу если интересно спросите. Но мне не нравится как там реализовано подключение кнопок и то что проекта нету на гитхаб, да и вообще нету какого то сообщества вокруг проекта и сайта где это публикуется) ну и там были баги с конектом, но говорят что это даже исправили. А Gimx вроде открытый проект и есть на гитхаб.

**Comment by lazenyuk-dmitry** [comment]
А так я с помощью Gimx играл в Forza horizon 4 и dirt rally. Вот не помню пробовал ли в project cars играть с Gimx. В city car driving у меня почему кнопка 32 не работает а на ней у меня вторая передача) тоже играл с Gimx.
